src/ds/address.h

```diff
@@ -132,6 +157,12 @@ namespace snmalloc
 #endif
   }
 
+  template<typename T = void>
```

Comment:
Doc comment?  Specifically, for these, explain why they are defined only on `AuthPtr` and not other pointer types.

Is there a reason that these are free-standing functions and not methods on `AuthPtr`?

Reply:
It's not a *good* reason, but, as to why they're freestanding: the AAL defines `address_t` and uses the `*Ptr` wrapping types in its own methods.  So far as I know, I can't declare a member of a class with a "trust me, there'll be a type for you by the time you care" signature.

Reply:
Please add that to a comment.